rmc/utils/constraint.py

```diff
@@ -955,214 +955,199 @@ def get_rescue_transcripts_and_create_no_breaks_ht(overwrite: bool) -> None:
+    )
+    logger.info(
+        "Single search round numbers: %s\nSimultaneous search round numbers: %s",
+        ",".join(map(str, single_search_round_nums)),
+        ",".join(map(str, simul_search_round_nums)),
+    )
+    if single_search_round_nums != simul_search_round_nums:
+        raise DataException(
+            "Round numbers from single and simultaneous searches do not match, please double-check!"
         )
-        ht = ht.filter(ht.locus.position <= ht.break_pos[0])
-        ht = ht.annotate(
-            # Start position is transcript start if this is the first section
-            section_start_pos=ht.start_pos,
-            section_end_pos=ht.break_pos[0],
+    if len(single_search_round_nums) == 0:
+        raise DataException("No search rounds recorded, please double-check!")
+    if len(single_search_round_nums) == 1:
+        logger.warning(
+            "Only one break search round recorded. Either no breaks were found or break search is not complete, please double-check!"
         )
-    elif section_type == "middle":
-        ht = ht.filter(
-            (ht.locus.position > ht.break_pos[indices[0]])
-            & (ht.locus.position <= ht.break_pos[indices[1]])
+    if single_search_round_nums != list(range(1, max(single_search_round_nums))):
+        raise DataException(
+            "Search round numbers are not consecutive and increasing from 1, please double-check!"
         )
+    return single_search_round_nums
 
-        # Add section start/end position annotations
-        ht = ht.annotate(
-            # Add 1 to break_pos[indices[0]] since it isn't actually included in the region
-            section_start_pos=ht.break_pos[indices[0]] + 1,
-            section_end_pos=ht.break_pos[indices[1]],
-        )
 
-    else:
-        logger.info(
-            "Getting info for last section of transcript (after largest breakpoint pos)..."
-        )
-        ht = ht.filter(ht.locus.position > ht.break_pos[-1])
-        ht = ht.annotate(
-            # Get last position from break_pos list and add 1 since it isn't included in the region
-            # Use the transcript end position as the end position since this is the last section
-            section_start_pos=ht.break_pos[-1] + 1,
-            section_end_pos=ht.end_pos,
-        )
+def merge_round_no_break_ht(is_rescue: bool, search_num: int) -> hl.Table:
+    """
+    Get merged single and simultaneous search no-break table from a given round of break search.
 
-    ht = ht.annotate(section=hl.format("%s_%s", ht.transcript, str(section_num)))
-    ht = get_subsection_exprs(
-        ht, "transcript", "observed", "mu_snp", "total_mu", "total_exp"
+    :param is_rescue: Whether to operate on search in rescue pathway.
+    :param search_num: Search iteration number
+        (e.g., second round of searching for single break would be 2).
+
+    :return: Table of loci in sections where no breaks were found in the break search round.
+    """
+    ht = hl.read_table(
```

Comment:
this HT has extraneous annotations (e.g., `reverse`) -- it might be good to have an annotations arg that selects only the relevant annotations (this will also help the `union` below, since that fails if the fields in the tables being unioned don't match)

Reply:
Addressed